frontend/packages/dev-console/src/components/pipelines/pipeline-builder/validation-utils.ts

```diff
@@ -1,51 +1,185 @@
 import * as yup from 'yup';
```

Comment:
Open to any and all feedback for what I did here. I think I got the only implementation I could ... but there was some cryptic comments from the owner of `yup` on issues that I didn't find a lot of documentation on. "Context" apparently is somehow I can do this... but I couldn't find a good use-case on the web, the docs or in searched issues.

I'll likely export these functions when I go to write tests.

Reply:
Seems to be fine on the first look. Will pull your PR locally and go through it in details. Need to get a better understanding of the whole pipeline builder form and pipeline builder code in general.

Reply:
kk, if you want a walkthrough / bjn conversation, do let me know. Happy to close the knowledge gap if desired.